upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -557,6 +562,58 @@ func deleteInstance(c AWSCloud, i *cloudinstances.CloudInstance) error {
 	return nil
 }
 
+// deregisterInstanceFromClassicLoadBalancer ensures that connectionDraining completes for the associated loadBalancer to ensure no dropped connections.
+// if instance is associated with an NLB, this method no-ops.
+func deregisterInstanceFromClassicLoadBalancer(c AWSCloud, i *cloudinstances.CloudInstance) error {
+	asg := i.CloudInstanceGroup.Raw.(*autoscaling.Group)
+
+	asgDetails, err := c.Autoscaling().DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: []*string{asg.AutoScalingGroupName},
+	})
+
+	if err != nil {
+		return fmt.Errorf("error describing autoScalingGroups: %v", err)
+	}
+
+	if len(asgDetails.AutoScalingGroups) == 0 {
+		return nil
+	}
+
+	// there will always be only one loadBalancer in the response.
+	loadBalancerNames := asgDetails.AutoScalingGroups[0].LoadBalancerNames
```

Comment:
Why will there only ever be one loadBalancer? I think we should handle the case in which there are multiple

Reply:
Typo in the comment; should say `there will always be only one ASG in the response from describeAutoScalingGroups`. I'll update.